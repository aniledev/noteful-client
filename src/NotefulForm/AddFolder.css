.AddFolder {
  background-color: #1d3656;
  color: #dedede;
  padding: 16px;
  max-width: 332px;
}

.AddFolder label {
  display: block;
  font-size: 16px;
  padding-bottom: 8px;
}

.AddFolder input,
.AddFolder textarea,
.AddFolder select {
  font-size: 14px;
  padding: 8px;
  width: 300px;
}

.AddFolder select {
  height: 33px;
}

.AddFolder > .field {
  margin-bottom: 16px;
}
.AddFolder > .buttons {
  margin-top: 26px;
}

.AddFolder button {
  background-color: #010d1e;
  border-radius: 4px;
  border: 0;
  color: #cacaca;
  display: block;
  font-size: 18px;
  font-weight: bold;
  padding: 8px;
  text-align: center;
  text-decoration: none;
}
.AddFolder button[type="submit"]:hover {
  background-color: #29862e;
  color: white;
}
