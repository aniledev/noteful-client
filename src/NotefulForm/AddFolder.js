import React, { Component } from "react";
import "./AddFolder.css";
import ValidationError from "../ValidationError";
import apiContext from "../apiContext";

export default class AddFolder extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: {
        value: "",
        changed: false
      }
    };
  }

  static contextType = apiContext;

  updateFolderName(name) {
    this.setState({
      name: { value: name, changed: true }
    });
  }

  validateFolderName() {
    const name = this.state.name.value.trim();
    if (name.length === 0) {
      return "A folder name is required.";
    } else if (name.length < 3) {
      return "Folder name must be at least 3 characters long.";
    }
  }

  handleSubmit(event) {
    event.preventDefault();
    const { name } = this.state;
    console.log(`Name: ${name.value}`);

    const myHeaders = new Headers();
    myHeaders.append("Content-Type", "application/json");

    const JSONBody = JSON.stringify({ name: name.value });

    const options = {
      method: "POST",
      headers: myHeaders,
      body: JSONBody,
      redirect: "follow"
    };

    fetch("http://localhost:9090/folders", options)
      .then(res => {
        if (!res.ok) {
          throw new Error("Something went wrong. Try again later.");
        } else if (res.ok) {
          if (!res.redirected) {
            window.location.href = "http://localhost:3000/";
          }
        }
      })
      .catch(error => alert(error.message));
  }

  render() {
    const folderNameError = this.validateFolderName();

    return (
      <form className="AddFolder" onSubmit={e => this.handleSubmit(e)}>
        <h2>Add a new folder</h2>
        <div className="form-group">
          <label>Folder Name</label>
          <input
            type="text"
            className="name-input"
            name="name"
            id="name"
            defaultValue="New Folder"
            onChange={e => this.updateFolderName(e.target.value)}
          />
        </div>
        {this.state.name.changed && (
          <ValidationError message={folderNameError} />
        )}

        <button
          type="submit"
          className="name-button"
          disabled={this.validateFolderName()}
        >
          Add Folder
        </button>
      </form>
    );
  }
}
