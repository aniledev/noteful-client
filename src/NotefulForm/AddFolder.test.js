import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import AddFolder from "./AddFolder";

describe(`AddFolder component`, () => {
  const props = {
    className: "test-class-name",
    children: <p>test children</p>,
    "data-other": "test-other-prop"
  };

  it("renders a form.AddFolder by default", () => {
    const wrapper = shallow(<AddFolder />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the AddFolder given props", () => {
    const wrapper = shallow(<AddFolder {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
