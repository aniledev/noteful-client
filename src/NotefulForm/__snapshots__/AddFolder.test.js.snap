// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFolder component renders a form.AddFolder by default 1`] = `
<form
  className="AddFolder"
  onSubmit={[Function]}
>
  <h2>
    Add a new folder
  </h2>
  <div
    className="form-group"
  >
    <label>
      Folder Name
    </label>
    <input
      className="name-input"
      id="name"
      name="name"
      onChange={[Function]}
      placeholder="New Folder"
      type="text"
    />
  </div>
  <button
    className="name-button"
    disabled="A folder name is required."
    type="submit"
  >
    Add Folder
  </button>
</form>
`;

exports[`AddFolder component renders the AddFolder given props 1`] = `
<form
  className="AddFolder"
  onSubmit={[Function]}
>
  <h2>
    Add a new folder
  </h2>
  <div
    className="form-group"
  >
    <label>
      Folder Name
    </label>
    <input
      className="name-input"
      id="name"
      name="name"
      onChange={[Function]}
      placeholder="New Folder"
      type="text"
    />
  </div>
  <button
    className="name-button"
    disabled="A folder name is required."
    type="submit"
  >
    Add Folder
  </button>
</form>
`;
